stdlib/2and3/argparse.pyi

```diff
@@ -1,41 +1,82 @@
 # Stubs for argparse (Python 2.7 and 3.4)
 
 from typing import (
-    Any, Callable, Iterable, List, IO, Optional, Sequence, Tuple, Type, Union,
-    TypeVar, overload
+    Any, Callable, Dict, Generator, Iterable, List, IO, NoReturn, Optional,
+    Pattern, Sequence, Tuple, Type, Union, TypeVar, overload
 )
 import sys
 
 _T = TypeVar('_T')
+_ActionVar = TypeVar('_ActionVar', bound='Action')
 
 if sys.version_info >= (3,):
     _Text = str
 else:
     _Text = Union[str, unicode]
 
-ONE_OR_MORE = ...  # type: str
-OPTIONAL = ...  # type: str
-PARSER = ...  # type: str
-REMAINDER = ...  # type: str
-SUPPRESS = ...  # type: str
-ZERO_OR_MORE = ...  # type: str
+ONE_OR_MORE: str
+OPTIONAL: str
+PARSER: str
+REMAINDER: str
+SUPPRESS: str
+ZERO_OR_MORE: str
+_UNRECOGNIZED_ARGS_ATTR: str
 
 class ArgumentError(Exception): ...
 
-class ArgumentParser:
-    prog = ...  # type: _Text
-    usage = ...  # type: Optional[_Text]
-    description = ...  # type: Optional[_Text]
-    epilog = ...  # type: Optional[_Text]
-    formatter_class = ...  # type: Type[HelpFormatter]
-    prefix_chars = ...  # type: _Text
-    fromfile_prefix_chars = ...  # type: Optional[_Text]
-    argument_default = ...  # type: Optional[_Text]
-    conflict_handler = ...  # type: _Text
-    add_help = ...  # type: bool
+class _AttributeHolder:
+    def _get_kwargs(self) -> List[Tuple[str, Any]]: ...
+    def _get_args(self) -> List[Any]: ...
+
+class _ActionsContainer:
+    description: Optional[_Text]
+    prefix_chars: _Text
+    argument_default: Optional[_Text]
+    conflict_handler: _Text
+
+    _registries: Dict[_Text, Dict[Any, Any]]
+    _actions: List[Action]
+    _option_string_actions: Dict[_Text, Action]
+    _action_groups: List[_ArgumentGroup]
+    _mutually_exclusive_groups: List[_MutuallyExclusiveGroup]
+    _defaults: Dict[str, Any]
+    _negative_number_matcher: Pattern[str]
+    _has_negative_number_optionals: List[bool]
+
+    def __init__(self, description: Optional[_Text], prefix_chars: _Text,
+                 argument_default: Optional[_Text], conflict_handler: _Text) -> None: ...
+    def register(self, registry_name: _Text, value: Any, object: Any) -> None: ...
+    def _registry_get(self, registry_name: _Text, value: Any, default: Any = ...) -> Any: ...
+    def set_defaults(self, **kwargs: Any) -> None: ...
+    def get_default(self, dest: _Text) -> Any: ...
+    def add_argument(self, *args: Any, **kwargs: Any) -> Action: ...
```

Comment:
Why did you remove the more specific arguments to this function in the previous stub?

Reply:
Those were incorrect.  I'm actually not entirely sure where they came from, given that in `argparse.py`, `add_argument` is actually defined using *args and **kwargs.

Reply:
The Python-visible signature is *args, **kwargs, but obviously in practice this method doesn't take any argument. Since it's probably the most frequently called part of argparse's interface, I think it's important that we give more specific types where it's at all possible. The documentation (https://docs.python.org/3.7/library/argparse.html#argparse.ArgumentParser.add_argument) also gives more specific arguments.